cmd/bb_scheduler/main.go

```diff
@@ -34,7 +34,20 @@ func main() {
 		log.Fatal(http.ListenAndServe(schedulerConfiguration.MetricsListenAddress, nil))
 	}()
 
-	executionServer, schedulerServer := builder.NewWorkerBuildQueue(util.DigestKeyWithInstance, schedulerConfiguration.JobsPendingMax)
+	var execDigestFunc remoteexecution.DigestFunction
+	switch schedulerConfiguration.ExecutionDigestFunction {
+	case "sha256":
```

Comment:
For this couldn't you use rather the [facilities provided by protobuf](https://developers.google.com/protocol-buffers/docs/reference/go-generated#enum)? For reference, you could look at [the generated code](https://github.com/bazelbuild/remote-apis/blob/master/build/bazel/remote/execution/v2/remote_execution.pb.go#L68). This would avoid using strings here.

Reply:
Or: Instead of using a string, you could use the enum type directly inside the .proto file. That way they are guaranteed to remain in sync.

Reply:
yup, I'm trying to pull in the remote-apis enum . It's resisting but I hope to subdue it soon.